pkg/resources/nodeagent/configsecret.go

```diff
@@ -317,10 +317,11 @@ func (n *nodeAgentInstance) generateUpstreamNode(index int32) upstreamNode {
 	podName := n.FluentdQualifiedName(fmt.Sprintf("%s-%d", fluentd.ComponentFluentd, index))
 	return upstreamNode{
 		Name: podName,
-		Host: fmt.Sprintf("%s.%s.%s.svc.cluster.local",
+		Host: fmt.Sprintf("%s.%s.%s.svc.%s",
```

Comment:
I think this format string will have the same problem with the trailing dot if `ClusterDomain` is set to empty as I noted earlier with the chart template.